### PR TITLE
Adds filter by PR to Actions mode (Fixes #205)

### PR DESCRIPTION
**Pull request by acoliver**
## Summary

Add a **filter by PR** option to the Actions mode filter bar (congruent with the existing workflow/status filters). When active, the run list shows only runs whose event is pull_request and that correspond to a specific PR number.

This also adds a **cross-mode action**: pressing g/G in PR mode jumps to Actions mode pre-filtered to the currently focused PR runs.

### Domain layer
- **ActionsFilter** gains pr_number: Option<u64> (display) and head_sha: Option<String> (API head_sha= param) fields, mirroring the existing workflow/workflow_path pattern
- **PullRequest** and **PullRequestDetail** gain head_sha: String to carry the PR head commit SHA (fetched from headRefOid)

### GitHub API boundary
- The actions/runs query now appends event=pull_request and head_sha=<sha> when the PR filter is active
- The PR GraphQL search queries and gh pr view --json field list now include headRefOid so the SHA is available for Actions filtering

### State/reducer
- ActionsFilterField gains a Pr variant (field index 2 in the filter bar)
- cycle_pr_filter() cycles the draft filter through the loaded PRs from PRs mode
- enter_actions_mode_with_pr_filter() sets both committed and draft filters on entry
- Filter navigation wraps across 3 fields (workflow, status, pr)

### Input routing
- The filter bar Delete/ClearCurrent handler now clears the PR field
- CycleFilterStatus at field index 2 cycles through PRs
- New g/G key handler in PR mode resolves the current PR SHA (from detail or list) and emits EnterActionsModeWithPrFilter

### UI rendering
- The Actions filter bar now renders a third field: pr with display value #NUM or any
- ACTIONS_FIELDS_PER_ROW updated to 3

### Tests
- Unit tests for build_runs_api_path with/without PR filter
- Unit tests for cycle_pr_filter (forward cycling, empty PRs noop, clear field)
- Unit tests for enter_actions_mode_with_pr_filter (committed + draft set)
- Unit tests for g/G cross-mode action (from detail, from list, with no PR, with composer open)
- Unit tests for filter navigation wrapping across 3 fields
- Tests for headRefOid parsing in PR list and detail JSON fixtures
- TUI scenario: dev-docs/tmux-scenarios/actions-pr-filter.json

## Test plan

- [x] cargo fmt --all --check — clean
- [x] cargo clippy --workspace --all-targets --all-features -- -D warnings — clean
- [x] cargo build --workspace --all-features --locked — compiles
- [x] cargo test --workspace --all-features — all tests pass
- [x] make ci-check — green (fmt, clippy, coverage >30%, build, test)

Closes #205

### CODE REVIEW

File: src/app_input/prs.rs [hunk 1]
```diff
@@ -114,6 +117,30 @@ fn resolve_pr_global_key(state: &AppState, key_event: &KeyEvent) -> Option<AppEv
...
+            return AppEvent::EnterActionsModeWithPrFilter {
+                pr_number: pr.number,
+                head_sha: pr.head_sha.clone(),
+            };
+        }
+        AppEvent::EnterActionsMode
+    } else {
+        AppEvent::EnterActionsMode
+    }
+}
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/app_input/prs.rs:161]: Fixed in commit ec26952. The `pr_to_actions_event` function now prefers the selected list PR as authoritative. `pr_detail` is only used when its number matches the selected list item (to benefit from the detail fetch's potentially fresher SHA). When the selection has moved away from the detail, the list item's SHA is used. A test (`g_prefers_selected_pr_over_stale_detail`) was added to cover this edge case.
